From 95622c4261ef6e0eb6688196d87d3bed838aa91d Mon Sep 17 00:00:00 2001
From: Sebastian Ott <sebott@redhat.com>
Date: Wed, 29 Oct 2025 17:59:07 +0100
Subject: [PATCH 08/10] x86: create new rhel 10.2 specific pc-q35 machine type

RH-Author: Sebastian Ott <sebott@redhat.com>
RH-MergeRequest: 416: x86, arm: create new rhel 9.8, 10.2 specific machine types
RH-Jira: RHEL-105826 RHEL-105828
RH-Acked-by: Eric Auger <eric.auger@redhat.com>
RH-Acked-by: Gavin Shan <gshan@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Commit: [3/4] 7f5a5fa45696e579cd08fb3869884ce9543badd5 (seott1/cos-qemu-kvm)

Signed-off-by: Sebastian Ott <sebott@redhat.com>
---
 hw/i386/pc_q35.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index e5d10c0335..aea4735dd3 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -690,10 +690,20 @@ DEFINE_Q35_MACHINE(2, 6);
 
 /* Red Hat Enterprise Linux machine types */
 
-static void pc_q35_rhel_machine_10_0_0_options(MachineClass *m)
+static void pc_q35_rhel_machine_10_2_0_options(MachineClass *m)
 {
     PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
     pc_q35_machine_options(m);
+    m->desc = "RHEL-10.2.0 PC (Q35 + ICH9, 2009)";
+    pcmc->smbios_stream_product = "RHEL";
+    pcmc->smbios_stream_version = "10.2.0";
+}
+DEFINE_Q35_MACHINE_AS_LATEST(10, 2, 0);
+
+static void pc_q35_rhel_machine_10_0_0_options(MachineClass *m)
+{
+    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
+    pc_q35_rhel_machine_10_2_0_options(m);
     m->desc = "RHEL-10.0.0 PC (Q35 + ICH9, 2009)";
     pcmc->smbios_stream_product = "RHEL";
     pcmc->smbios_stream_version = "10.0.0";
@@ -707,7 +717,7 @@ static void pc_q35_rhel_machine_10_0_0_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_rhel_10_1_compat,
                      pc_rhel_10_1_compat_len);
 }
-DEFINE_Q35_MACHINE_AS_LATEST(10, 0, 0);
+DEFINE_Q35_MACHINE(10, 0, 0);
 
 static void pc_q35_rhel_machine_9_6_0_options(MachineClass *m)
 {
-- 
2.47.3

